src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -283,7 +506,146 @@ static bool areStartUpOnOffServerAttributesTokenized(EndpointId endpoint)
 
     return true;
 }
-#endif
+#endif // ZCL_USING_ON_OFF_CLUSTER_START_UP_ON_OFF_ATTRIBUTE
+
+/**
+ * @brief event control object for an endpoint
+ *
+ * @param[in] endpoint
+ * @return EmberEventControl* configured event control
+ */
+EmberEventControl * OnOffServer::getEventControl(EndpointId endpoint)
+{
+    uint16_t index = emberAfFindClusterServerEndpointIndex(endpoint, OnOff::Id);
+    return &eventControls[index];
```

Comment:
This can index out of the array, especially if the endpoint is dynamic.  This needs a check on the index value!

Reply:
Fixed in #12562